addons/sale_timesheet/static/src/components/so_line_field/so_line_field.js

```diff
@@ -11,7 +11,10 @@ export class SoLineField extends Many2OneField {
         const update = this.update;
         this.update = (value, params = {}) => {
             update(value, params);
-            if (value || this.updateOnEmpty) {
+            if ( // field is unset AND the old & new so_lines are different
+                !this.props.record.data.is_so_line_edited &&
+                this.props.value[0] != (value && value[0].id)
```

Comment:
Does it work if the value in the props is `false` (= unset)? 🧐

Reply:
It doesn't throw an error because `false[0] == undefined`, but I guess if the value is empty before and after the update, it will end up comparing `undefined != false` which is `true`.

To prevent that I could either do `(this.props.value && this.props.value[0]) != (value && value[0].id)` or `this.props.value[0] != (value[0] && value[0].id)` to make sure both are equally falsy when empty.

I reckon the second one is clearer, even though it relies on a JS quirk.

Reply:
Okay, I see thanks for your explanation. 🙂